libmscore/sym.cpp

```diff
@@ -6586,6 +6591,69 @@ void initScoreFonts()
       QFont::insertSubstitution("Finale Broadway Text",  "MuseJazz Text");
       QFont::insertSubstitution("ScoreFont",      "Leland Text"); // alias for current Musical Text Font
       ScoreFont::fallbackFont();   // load fallback font
+
+      QString userFontsPath = preferences.getString(PREF_APP_PATHS_MYSCOREFONTS);
+      scanUserFonts(userFontsPath);
+      preferences.addOnSetListener([](const QString& key, const QVariant& value) {
+            if (key == PREF_APP_PATHS_MYSCOREFONTS) {
+                  scanUserFonts(value.toString());
+                  }
+            });
+      }
+
+void ScoreFont::scanUserFonts(const QString& path)
+      {
+      QVector<ScoreFont> userfonts;
+
+      QDirIterator iterator(path, QDir::Dirs | QDir::NoDotAndDotDot | QDir::Readable);
+
+      while (iterator.hasNext()) {
+          QString fontDir = iterator.next();
+          QString fontDirPath = iterator.filePath() + "/";
+          QString fontDirName = iterator.fileName();
+
+          QString fontName;
+          QString fontFilename;
+          QDirIterator innerIterator(fontDirPath, { "*.otf", "*.ttf" }, QDir::Files);
+
+          while (innerIterator.hasNext()) {
+              QString potentialFontFile = innerIterator.next();
+              QFileInfo fileinfo(potentialFontFile);
+
+              if (fileinfo.completeBaseName().toLower() == fontDirName.toLower()) {
+                  fontName = fileinfo.completeBaseName();
+                  fontFilename = innerIterator.fileName();
+                  break;
+              }
+          }
+
+          bool hasMetadataFile = QFileInfo::exists(fontDirPath + "metadata.json");
```

Comment:
Would be nice to allow for `<fontname>_metatada.json' here (but not making it mandatory), or simply `*metadata.json`, or even `*.json`, at least if there is only one?

Reply:
Dynamics do work for me, but not using those from Finale Ash or Leipzig, seems to fall back to Bravura (Text)

Reply:
It might be you're renaming away the spaces in the filenames. It probably needs the space before the "Text" for the text fonts

Reply:
See https://github.com/Jojo-Schmitz/MuseScore/pull/82/files#diff-9e3816e63546943283fc7dc1cebdb8edfd0d1ca4bb2f0713a3cfe607308a5ff4R3179-R3180